feat(internal): extend messenger-api

buscape opened the pull request:
This PR extends `@userlike/messenger` v1-api in with regard to authentication and re-authentication in a non-breaking fashion.
As of today those changes are intended for internal usage. Issue: optixx/Userlike/issues/21689

## types (added)
Types for Credentials and MountOptions:
```typescript
interface Credentials {
  token: string;
  uuid: string;
}

interface MountOptions {
  credentials?: Credentials;
  openConversationId?: number;
}
```

## mount (updated)
Optional MountOptions:
```typescript
  /**
   * Create the messenger.
   * When credentials are provided those will be used for authentication.
   * When openConversationId is passed that conversation will be opened after successful authentication.
   */
  mount(opts?: MountOptions): Promise<ActionResult<string, void>>;
```

## consumeToken (unstable/experimental)
Unstable for now so we can test the API internally first.
```typescript
  /**
   * Experimental.
   * Consumes a short living token to returns long living AuthParams.
   */
   __unstableConsumeToken(token: string): Promise<ActionResult<string, Credentials>>;
```
## openConversation (unstable/experimental)
Unstable for now so we can test the API internally first.
```typescript
/**
 * Experimental.
 * Opens a conversation by id.
 */
__unstableOpenConversation(
    id: number
  ): Promise<ActionResult<string, void>>;
```

Commit:
chore: add .vscode directory to gitignore

Commit:
feat: add consumeToken to v1 api

Commit:
feat: add support for an optional credentials param to v1 api mount

Commit:
feat: add experimental __unstableOpenConversation to v1 api

Commit:
docs: add doc comment for setVisibilty

anilanar commented on packages/messenger-internal/src/v1/types.ts:
```diff
@@ -10,17 +10,32 @@ export interface ApiState {
   getState(): State;
 }
 
+export interface AuthParams {
```
Not 100% about this name but ok we can decide later.

anilanar committed:
Publish

 - @userlike/messenger-internal@2.2.0-alpha.0
 - @userlike/messenger@1.2.3-alpha.0

anilanar committed:
chore: empty commit to fix borked release

anilanar committed:
Publish

 - @userlike/messenger-internal@2.2.0-alpha.1
 - @userlike/messenger@1.2.3-alpha.1

Commit:
feat: add openConversationId to mount options

Commit:
refactor: rename AuthParams to Credentials, introduce MountOptions

Commit:
refactor: make consumeToken an experimental __unstable api method for now

Commit:
chore: improve doc-comments

anilanar commented on packages/messenger-internal/src/v1/types.ts:
```diff
@@ -10,17 +10,32 @@ export interface ApiState {
+
+export interface MountOptions {
+  credentials?: Credentials;
+  openConversationId?: number;
```
If we are able to `api.mount().then(() => api.__unstableOpenConversation(id))`, then let's remove this param from options.

anilanar commented on packages/messenger-internal/src/v1/types.ts:
```diff
@@ -57,6 +72,12 @@ export interface ApiActions {
 
+  /**
+   * Experimental.
+   * Consumes a short living token to returns long living AuthParams.
```
Still mentions `AuthParams`.

Commit:
fix: remove mention of AuthParams

Commit:
refactor: remove openConversationId from MountOptions

anilanar committed:
Publish

 - @userlike/messenger-internal@2.2.0-alpha.2
 - @userlike/messenger@1.2.3-alpha.2
